src/schema/schema_stat.c

```diff
@@ -53,6 +53,60 @@ err:	__wt_scr_free(session, &buf);
 }
 
 /*
+ * __curstat_size_only --
+ *	 For very simple tables we can avoid getting table handles if
+ *	 configured to only retrieve the size. It's worthwhile because
+ *	 workloads that create and drop a lot of tables can put a lot of
+ *	 pressure on the table list lock.
+ */
+static int
+__curstat_size_only(WT_SESSION_IMPL *session,
+    const char *uri, bool *was_fast,WT_CURSOR_STAT *cst)
+{
+	WT_CONFIG cparser;
+	WT_CONFIG_ITEM ckey, colconf, cval;
+	WT_DECL_RET;
+	WT_ITEM namebuf;
+	wt_off_t filesize;
+	char *tableconf;
+
+	WT_CLEAR(namebuf);
+	*was_fast = false;
+
+	if (!F_ISSET(cst, WT_CONN_STAT_SIZE))
+		return (0);
+
+	WT_RET(__wt_metadata_search(session, uri, &tableconf));
+	WT_ERR(__wt_config_getones(session, tableconf, "columns", &colconf));
+
+	WT_ERR(__wt_config_subinit(session, &cparser, &colconf));
+	if ((ret = __wt_config_next(&cparser, &ckey, &cval)) == 0)
+		goto err;
```

Comment:
This could use a comment: "Checking for a simple table with no column names.  Otherwise, there may be multiple sources involved and we have to take the slow path."